### PR TITLE
Add `FloorTest` and clean-up `Floor`

### PR DESCRIPTION
**Pull request by ricardo-ramos-moura**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/test/java/com/thealgorithms/maths/FloorTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    public void testFloorWholeNumber() {
+        assertEquals(0, Floor.floor(0));
+        assertEquals(1, Floor.floor(1));
+        assertEquals(-1, Floor.floor(-1));
+        assertEquals(42, Floor.floor(42));
+        assertEquals(-42, Floor.floor(-42));
+    }
+
+    @Test
+    public void testFloorDecimalNumber() {
```
Comment (vil02) [inline src/test/java/com/thealgorithms/maths/FloorTest.java:17]: The word `decimal` is not correct here. `Floor.floor` is defined for `double`.

**Review by vil02 — CHANGES_REQUESTED** [review]
Could you please also _tidy up_ the `Floor` class, namely:
- remove the `main` method,
- mark the class as `final`,
- add a `private` default constructor of `Floor`, i.e.
    ```java
    private Floor() {
    }
    ```

File: src/test/java/com/thealgorithms/maths/FloorTest.java (hunk 1 above)
Comment (ricardo-ramos-moura) [reply src/test/java/com/thealgorithms/maths/FloorTest.java:17]: @vil02  Implemented changes as requested.

**Review by vil02 — APPROVED** [review]

**Comment by vil02** [comment]
@siriak please have a look regarding changes in `DIRECTORY.md`.

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**